Yank/yank.py

```diff
@@ -218,7 +218,7 @@ def resume(self, phases=None):
         # If no phases specified, construct a list of phases from the filename prefixes in the store directory.
         if phases is None:
             phases = utils.find_phases_in_store_directory(self._store_directory)
-        self._phases = phases.keys()
+        self._phases = sorted(phases.keys())  # sorting ensures all MPI processes run the same phase
```

Comment:
Good idea! I guess we'll have to switch to this solution if we'll ever want to force a particular order. For now, I'd say this code is simpler. Also, it may be refactored during the API overhaul with the introduction of single-phase `AlchemicalPhase` objects.

I'll merge as soon as tests pass then!